sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/HttpUtil.java

```diff
@@ -75,6 +65,24 @@ public static String getUserAgentPrefix() {
                        .orElse(DEFAULT_USER_AGENT_VALUE_PREFIX);
     }
 
+    public static String post(String url, Map<String, String> headers, String body, String contentType) {
+        String result = null;
+        try (CloseableHttpClient client = HttpClients.createDefault()) {
+            HttpPost httpPost = new HttpPost(url);
+            if (headers != null) {
+                httpPost.addHeader(USER_AGENT_KEY, USER_AGENT_VALUE);
```

Comment:
If headers == null, we still need `httpPost.addHeader(USER_AGENT_KEY, USER_AGENT_VALUE);`.